2.train-VAE/scripts/5.apply_ica.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Apply ICA to gene dependencies
+# 
+# Independent Component Analysis (ICA) is a dimensionality reduction method that separates a multivariate signal into additive, independent components.
+# 
+# We apply it to GeneEffect scores here and save latent representations (ICA components) for downstream comparative analyses (to compare with BetaVAE).
+# 
+
+# In[1]:
+
+
+import sys
+import pathlib
+import pandas as pd
+import plotnine as gg
+from sklearn.decomposition import FastICA
+
+sys.path.insert(0, "../utils/")
+from data_loader import load_model_data
+
+
+# In[2]:
+
+
+data_directory = pathlib.Path("../0.data-download/data")
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet")
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet")
+
+output_dir = pathlib.Path("results")
+ica_output_file = pathlib.Path(f"{output_dir}/ica_latent.parquet.gz")
+output_ica_weights_file = pathlib.Path(f"{output_dir}/ICA_weight_matrix_gsea.parquet")
+
```

Comment:
Consider resolving thsese paths and ensuring that input paths are in fact present to raise errors here instead of downstream
e.g. `dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet").resolve(strict=True)`

Reply:
Done